### PR TITLE
Tutorial on burn scar analysis using embeddings from partial inputs

### LINKED ISSUE #126: Test partial input to Clay v0
After some discussions, we realized that there is a technical possiblity to run the Clay v0 model with partial inputs. We need to 

- [ ] Create an example notebook to show how that can be done
- [ ] Test the model with a benchmark that compares partial input (ideally one for Sentinel-2 and one for Sentinel-1) with full inputs.
- [ ] Document this.

### LINKED ISSUE #136: Create script to make mini datacubes
This is to support stories about embeddings on specific use cases. The script should provide a light-weight way to generate data that can be fed to the model for exploring embeddings.

The first iteration of this will focus on collecting Sentinel-2 and Sentinel-1 datacubes, but separately, not aligned. These can then be passed to the model using partial inputs https://github.com/Clay-foundation/model/issues/126.

### PR DESCRIPTION
**Pull request by yellowcap**
This PR adds a tutorial on Burn scar analysis using embeddings from partial inputs

This notebook contains a complete example for how to run Clay. It combines the following three different aspects

1. Create single-chip datacubes with time series data for a location and a date range
2. Run the model with partial inputs, in this case RGB + NIR
3. Study burn scares through the embeddings generated for that datacube

Closes #126
Closes #136

### CODE REVIEW

File: docs/partial-inputs.ipynb [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "77e7c22c-1bfd-4281-bb12-8330c3eedc25",
+   "metadata": {},
+   "source": [
+    "## Write data to tif files\n",
+    "To use the mini datacube in the Clay dataloader, we need to write the\n",
+    "images to tif files on disk. These tif files are thenb used by the Clay\n",
```
Comment (weiji14) [inline docs/partial-inputs.ipynb:195]:
```suggestion
    "images to tif files on disk. These tif files are then used by the Clay\n",
```

File: docs/partial-inputs.ipynb [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+    "        1597.68,  # green\n",
+    "        1741.10,  # blue\n",
+    "        2858.43,  # nir\n",
+    "    ]\n",
+    "    STD = [\n",
+    "        2026.96,  # red\n",
+    "        2011.88,  # green\n",
+    "        2146.35,  # blue\n",
+    "        2016.38,  # nir\n",
+    "    ]\n",
```
Comment (weiji14) [inline docs/partial-inputs.ipynb:295]: Should we document the mean/std values for all the bands (not just RGB+NIR) somewhere for v0? At least until we resolve #94?

File: docs/partial-inputs.ipynb [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+     "text": [
+      "Average embeddings have shape (12, 768)\n"
+     ]
+    }
+   ],
+   "source": [
+    "embeddings = []\n",
+    "\n",
+    "for batch in trn_dl:\n",
+    "    with torch.no_grad():\n",
```
Comment (weiji14) [inline docs/partial-inputs.ipynb:334]: Prefer `torch.inference_mode`, see https://pytorch.org/docs/2.2/generated/torch.inference_mode.html#inference-mode
```suggestion
    "    with torch.inference_mode():\n",
```

File: docs/_toc.yml [hunk 4]
```diff
@@ -32,6 +32,8 @@ parts:
       file: model_embeddings
     - title: Finetuning
       file: model_finetuning
+    - title: Partial inputs
+      file: partial-inputs
 - caption: Tutorials
```
Comment (weiji14) [inline docs/_toc.yml:37]: Should this go under the tutorials section below?

File: docs/partial-inputs.ipynb [hunk 5]
```diff
@@ -0,0 +1,446 @@
...
+    "    items,\n",
+    "    bounds=bounds,\n",
+    "    snap_bounds=False,\n",
+    "    epsg=epsg,\n",
+    "    resolution=10,\n",
+    "    dtype=\"float32\",\n",
+    "    rescale=False,\n",
+    "    fill_value=0,\n",
+    "    assets=BAND_GROUPS[\"rgb\"] + BAND_GROUPS[\"nir\"],\n",
+    "    resampling=Resampling.bilinear,\n",
```
Comment (weiji14) [inline docs/partial-inputs.ipynb:170]: Should be using nearest neighbour resampling for optical imagery, see https://github.com/Clay-foundation/model/issues/21#issuecomment-1815256781.
```suggestion
    "    resampling=Resampling.nearest,\n",
```

File: docs/partial-inputs.ipynb [hunk 6]
```diff
@@ -0,0 +1,446 @@
...
+    ")\n",
+    "\n",
+    "stack = stack.compute()\n",
+    "\n",
+    "stack.sel(band=[\"red\", \"green\", \"blue\"]).plot.imshow(\n",
+    "    row=\"time\", rgb=\"band\", vmin=0, vmax=2000, col_wrap=7\n",
```
Comment (weiji14) [inline docs/partial-inputs.ipynb:176]: Since there are 12 images, could maybe use col_wrap=6? Or did you want the top row to be pre-fire, and bottom row to be post-fire images?

```suggestion
    "stack.sel(band=[\"red\", \"green\", \"blue\"]).plot.imshow(\n",
    "    row=\"time\", rgb=\"band\", vmin=0, vmax=2000, col_wrap=6\n",
```

File: docs/partial-inputs.ipynb [hunk 7]
```diff
@@ -0,0 +1,446 @@
...
+   "metadata": {},
+   "source": [
+    "## Analyze embeddings\n",
+    "Now we can make a simple analysis of the embeddings. We reduce all the\n",
+    "embeddings to a single number using Principle Component Analysis. Then\n",
+    "we can plot the principal components. The effect of the fire on the\n",
+    "embeddings is clearly visible.\n",
+    "\n",
+    "In the plot, each image embedding is one point. One can clearly distinguish\n",
+    "the two cloudy images and the values after the fire are consistently low.\n",
```
Comment (weiji14) [inline docs/partial-inputs.ipynb:370]: Maybe good to move this sentence to a new markdown cell below the scatter plot, to be more like a caption describing the plot. I'd also recommend adding a sentence to remind the viewer to refer to the RGB plots above see the cloudy and burn scar images.

File: docs/partial-inputs.ipynb [hunk 8]
```diff
@@ -0,0 +1,446 @@
...
+      "text/plain": [
+       "<Figure size 640x480 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "pca = decomposition.PCA(1)\n",
```
Comment (weiji14) [inline docs/partial-inputs.ipynb:409]: Be more explicit on what 1 means here.
```suggestion
    "pca = decomposition.PCA(n_components=1)\n",
```

**Review by weiji14 — COMMENTED** [review]
Thanks @yellowcap, mostly just minor suggestions. I'd re-run the notebooks again with the nearest-neighbour resampling to make sure the results don't change much. Also would be good if @lillythomas can check the PCA part since she's been working on that more recently.

File: docs/_toc.yml (hunk 4 above)
Comment (yellowcap) [reply docs/_toc.yml:37]: absolutely

File: docs/partial-inputs.ipynb (hunk 7 above)
Comment (yellowcap) [reply docs/partial-inputs.ipynb:370]: good idea - done

File: docs/partial-inputs.ipynb (hunk 2 above)
Comment (yellowcap) [reply docs/partial-inputs.ipynb:295]: Added docs for this in https://github.com/Clay-foundation/model/pull/152

**Comment by lillythomas** [comment]
This looks great @yellowcap good to see how you plotted the dates directly from the stack. That works well for single component scenarios, which should be largely sufficient for many stories. More complex ones might require somehting like what I did where I added the non-normalized dates to the batches such that embeddings can be labeled with dates in the plots.

Signal of the burn scars is clear. This looks good to merge. I'll incorporate some takeaways from this into my PR.

**Review by lillythomas — APPROVED** [review]
